### PR TITLE
Created broadcast receiver to check internet connection

### PR DESCRIPTION
**Pull request by VVMaxVV**
Created broadcastreceiver to check internet connection and make auto refresh when internet connection resumes

### CODE REVIEW

File: app/src/main/java/com/example/internetshop/presentation/activity/fragments/CategoriesFragment.kt [hunk 1]
```diff
@@ -38,6 +40,10 @@ class CategoriesFragment : BaseFragment() {
             viewModel = this@CategoriesFragment.viewModel
             lifecycleOwner = this@CategoriesFragment
         }
+        context?.registerReceiver(
```
Comment (darthoo) [inline app/src/main/java/com/example/internetshop/presentation/activity/fragments/CategoriesFragment.kt:43]: Please replace `context?` with `requireContext`

File: app/src/main/java/com/example/internetshop/presentation/activity/fragments/CategoriesFragment.kt [hunk 2]
```diff
@@ -96,4 +96,9 @@ class CategoriesFragment : BaseFragment() {
                 .actionCategoriesFragmentToProductsFromCategoryFragment(categoryName)
         findNavController().navigate(action)
     }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        context?.unregisterReceiver(networkBroadcast.get())
```
Comment (darthoo) [inline app/src/main/java/com/example/internetshop/presentation/activity/fragments/CategoriesFragment.kt:102]: The same. Use requireContext

**Review by darthoo — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/internetshop/presentation/activity/fragments/CategoriesFragment.kt (hunk 1 above)
Comment (VVMaxVV) [reply app/src/main/java/com/example/internetshop/presentation/activity/fragments/CategoriesFragment.kt]: fixed

File: app/src/main/java/com/example/internetshop/presentation/activity/fragments/CategoriesFragment.kt (hunk 2 above)
Comment (VVMaxVV) [reply app/src/main/java/com/example/internetshop/presentation/activity/fragments/CategoriesFragment.kt]: fixed

**Pull request merged**